lib/internal/util/inspect.js

```diff
@@ -893,7 +901,7 @@ function formatValue(ctx, value, recurseTimes, typedArray) {
       // a counter internally.
       const depth = ctx.depth === null ? null : ctx.depth - recurseTimes;
       const isCrossContext =
-        proxy !== undefined || !FunctionPrototypeSymbolHasInstance(Object, context);
+        isProxyValue || !FunctionPrototypeSymbolHasInstance(Object, context);
```

Comment:
It is possible to check for `context !== value`. That way there is no need for the extra variable :)